### DISCUSSION TITLE
ダークモードのサポートについて

### DISCUSSION THREAD

**Discussion by kskwtnk** [increments/qiita-discussions#111] [要望・アイデア・不具合]
皆さんのご意見を聞かせてください！

https://github.com/increments/qiita-discussions/discussions/103 ←こちらのDiscussionでも話にあった、ダークモードのサポートについて

まず「ダークモードが欲しい」というご意見は多くいただいています
我々としても実施したいのですが、全てのページに適用してのリリースはかなり実現が難しいです

ですが、以下のような条件付きであれば実現できるかもしれません

- ページによってダークモードとライトモードが混在する
    - 例えば
    - 記事ページ：ダークモード有り
    - 過去のAdvent Calendarページ：ダークモード無し
- OS設定に合わせて自動で切り替わるのみ
    - つまり、UI上でのスイッチャーの提供は無し

コメントにて、皆さんはどのスタンスか教えてください！
このDiscussionの中だけで決めることはできませんが、社内での開発優先度を決める際に「実際の声」として挙げさせていただきます :pray: 

1. 条件付きであってもダークモードが欲しい
    - どのページにダークモードが欲しいかも一緒に教えていただけると幸いです
1. 全てのページに適用できるようになってからダークモードを提供して欲しい
1. その他
    - そもそも求めていない、などの意見であっても歓迎です

**Comment by munierujp** [comment]
私のスタンスとしては、「条件付きであってもダークモードが欲しい」です。
特に、個人的によく見るページ（記事ページ、ユーザーページ、検索結果ページ、ホームなど）だけでもダークモードに対応されるとありがたいです。
ダークモードに対応しているGitHubでも、[GitHub Marketplace](https://github.com/marketplace)など一部のページは未対応なので、Advent Calendarなど一部のページが未対応でもいいと思います。

**Comment by ipeblb** [comment]
いちライトユーザーの意見ですが、「条件付きでいいのであれば嬉しい」くらいのスタンスです
個人のユースケース的には対応範囲はトップページ、記事ページ、(可能なら検索結果ページ)まであるとありがたいです。

ダークモードがあるとありがたい例として、例えば夜スマホでTwitter(ダークモード)などからリンク遷移した際、遷移先が白基調のページだとそれだけで眩しくて閉じてしまったりするケースが割とあります。
情報収集がてらTwitter経由でQiitaを開くことがちょくちょくあるので、対応してもらえれば少し記事を読む機会が増えるかなあ、という感覚です。

個人的には上記のケース以外では、ダークモードの有無で利用頻度はあまり変わらないと思いますが、よく使うアプリやサイトに関しては、対応しているものは基本的にダークモードで利用しているので対応してもらえればありがたいなと思います。

**Comment by yoshi389111** [comment]
部分的でもいいので、ダークモードが対応されるとうれしいです。

ただし、暗い画面から明るい画面に遷移すると非常にまぶしく感じるので、記事ページと、記事ページからよく遷移されるページは同時に対応してほしいと思います。

* 記事ページ
* ホーム
* トレンド
* 検索結果
* ユーザーページ（マイページ）

などはセットで対応してもらえたらと思います。

UIのスイッチャーも可能であれば欲しいですが、なくてもダークモード対応してくれればありがたいです。
あるいはダークモードではなく、いつでも黒基調でもよいのですけど（たぶん無理だと思いますが）

**Comment by simon-ritchie** [comment]
個人的には条件付きでも欲しく感じます・・・！

小出しでの対応でライトモードとページによって混在してもそこまで気にしません！

（もちろん少しずつサポート範囲が多くなったりすると嬉しいです。少し実装してみてダークモードの利用率などを取られて優先順位を調整されたり・・・とかでもいいかもしれません）

とりあえずは記事ページが個人的には一番欲しく、次いでトップページが欲しく感じております・・・！

**Comment by kskwtnk** [comment]
@munierujp @ipeblb @yoshi389111 @simon-ritchie 
みなさんご意見ありがとうございます！

期日や、具体的にどのページまでサポートする……と現段階では確約できないのですが、プロジェクトとして進めていこうと思っています！
先ほど[ブログ](https://blog.qiita.com/ui-update-plan-for-2022/)にも投稿したのですが、以下のようなスケジュールで各種UIアップデートをしていく予定です！

- 2022年1月下旬
    - フィードの記事カードを1列にします
- 2022年春頃
    - 記事カードの情報量を見直します
- 2022年春〜夏
    - 表示速度の改善を実施します
- 2022年中
    - HeaderのUIを見直す予定です
    - サイドバーのコンテンツを見直す予定です
- 未定
    - ダークモードの提供

**Comment by kskwtnk** [comment]
@munierujp @ipeblb @yoshi389111 @simon-ritchie 
ダークモードについてご意見をくださったみなさん、ありがとうございました！
本日、ベータ版としてダークモードを公開しました！

https://blog.qiita.com/article-page-dark-mode-beta/

現状は記事ページだけですが、各種フィードページやユーザーページなどもできるだけすぐに追従したいと考えています
小さいですが一歩前進しましたのでご共有です！

**Reply by simon-ritchie** [reply]
@xrxoxcxox 

ご対応ありがとうございます・・・！β版で利用させていただきます・・・！🙇‍♂️

**Comment by tanaka-asami415** [comment]
ご報告が遅れてしまいましたが、ダークモードを正式版としてリリースしました！✨
ダークモードについてご意見をくださったみなさま、ありがとうございました！

正式版では、モードの切替を行えるよう「テーマカラー設定」機能を追加しております！

詳細は下記ブログをご参照ください！✨
https://blog.qiita.com/release-theme/

本Discussionはクローズいたしますが、何かお気づきの点があればDiscussions等で教えていただけますと幸いです！🙇
